WORKSPACE

```diff
@@ -0,0 +1,112 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
+load("@bazel_tools//tools/build_defs/repo:git.bzl", "new_git_repository")
+
+# `clang`/`llvm`
+#
+# Normally Bazel will use the system's version of clang as the default C
+# toolchain. This prevents the builds from being hermetic though so we include
+# our own.
+#
+# All of the clang related tools are provided under the `@llvm_toolchain_llvm`
+# repo. To see what's available run `bazel query @llvm_toolchain_llvm//...`.
+
+TOOLCHAINS_LLVM_VERSION = "1.0.0"
+TOOLCHAINS_LLVM_SHA256 = "e91c4361f99011a54814e1afbe5c436e0d329871146a3cd58c23a2b4afb50737"
+
+http_archive(
+    name = "toolchains_llvm",
+    sha256 = TOOLCHAINS_LLVM_SHA256,
+    strip_prefix = "toolchains_llvm-{0}".format(TOOLCHAINS_LLVM_VERSION),
+    canonical_id = "{0}".format(TOOLCHAINS_LLVM_VERSION),
+    url = "https://github.com/bazel-contrib/toolchains_llvm/releases/download/{0}/toolchains_llvm-{0}.tar.gz".format(TOOLCHAINS_LLVM_VERSION),
+)
+
+load("@toolchains_llvm//toolchain:deps.bzl", "bazel_toolchain_dependencies")
+bazel_toolchain_dependencies()
+
+load("@toolchains_llvm//toolchain:rules.bzl", "llvm_toolchain")
+llvm_toolchain(
+    name = "llvm_toolchain",
+    llvm_version = "16.0.0",
```

Comment:
The version we pick here is required to have a [GitHub release](https://github.com/llvm/llvm-project/releases) available for all of our required architectures. `v16.0.0` is the latest release that satisfies this :/ 

I would like to eventually define our own Bazel clang toolchain which would allow us to pick any version we want. But starting with this open source rule set was easier